provider/coindesk.go

```diff
@@ -0,0 +1,61 @@
+package provider
+
+import (
+	"crypto-price-tracker/config"
+	"crypto-price-tracker/responsestruct"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"time"
+)
+
+type CoinDeskResponse struct {
+	Time       TimeInfo `json:"time"`
+	Disclaimer string   `json:"disclaimer"`
+	ChartName  string   `json:"chartName"`
+	Bpi        BpiInfo  `json:"bpi"`
+}
+
+type TimeInfo struct {
+	Updated    string `json:"updated"`
+	UpdatedISO string `json:"updatedISO"`
+	UpdatedUK  string `json:"updateduk"`
+}
+
+type BpiInfo struct {
+	USD CurrencyInfo `json:"USD"`
+	EUR CurrencyInfo `json:"EUR"`
+	GBP CurrencyInfo `json:"GBP"`
+}
+
+type CurrencyInfo struct {
+	Code        string  `json:"code"`
+	Symbol      string  `json:"symbol"`
+	Rate        string  `json:"rate"`
+	Description string  `json:"description"`
+	RateFloat   float64 `json:"rate_float"`
+}
+
+func GetPricesFromCoinDeskAPI() responsestruct.ResponseStruct {
+	res, err := http.Get(config.CoinDeskAPIProviderURL)
+	config.LastFetchedAPITime = time.Now()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Log.fatal() is ```Fatal is equivalent to [Print] followed by a call to [os.Exit](1).```

So it will exit the execution completely, we should propagate the error from up wards (from client -> service -> transport -> middleware) and the middleware will output the error response to the API.